### ISSUE TITLE
resampled_image display issue

### ISSUE THREAD

**Issue by ShibaPrasad** [vincent1bt/Healthy-notebooks#1]
Followed the steps explained in notebook

I would like to know whay my resampled_image space showing (302, 302, 0). I believe that migh t be reason it's not displaying in  plt.imshow

resampled_image = resample(first_image, image_thickness, pixel_spacing)   
# array([], shape=(302, 302, 0), dtype=uint16)

np.squeeze(resampled_image).shape
**_(302, 302, 0)_**

**_plt.imshow(resampled_image)_**

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-43-eea311106858> in <module>
----> 1 plt.imshow(resampled_image)

~/anaconda3/lib/python3.7/site-packages/matplotlib/pyplot.py in imshow(X, cmap, norm, aspect, interpolation, alpha, vmin, vmax, origin, extent, shape, filternorm, filterrad, imlim, resample, url, data, **kwargs)
   2681         filternorm=filternorm, filterrad=filterrad, imlim=imlim,
   2682         resample=resample, url=url, **({"data": data} if data is not
-> 2683         None else {}), **kwargs)
   2684     sci(__ret)
   2685     return __ret

~/anaconda3/lib/python3.7/site-packages/matplotlib/__init__.py in inner(ax, data, *args, **kwargs)
   1599     def inner(ax, *args, data=None, **kwargs):
   1600         if data is None:
-> 1601             return func(ax, *map(sanitize_sequence, args), **kwargs)
   1602 
   1603         bound = new_sig.bind(ax, *args, **kwargs)

~/anaconda3/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py in wrapper(*args, **kwargs)
    367                 f"%(removal)s.  If any parameter follows {name!r}, they "
    368                 f"should be pass as keyword, not positionally.")
--> 369         return func(*args, **kwargs)
    370 
    371     return wrapper

~/anaconda3/lib/python3.7/site-packages/matplotlib/cbook/deprecation.py in wrapper(*args, **kwargs)
    367                 f"%(removal)s.  If any parameter follows {name!r}, they "
    368                 f"should be pass as keyword, not positionally.")
--> 369         return func(*args, **kwargs)
    370 
    371     return wrapper

~/anaconda3/lib/python3.7/site-packages/matplotlib/axes/_axes.py in imshow(self, X, cmap, norm, aspect, interpolation, alpha, vmin, vmax, origin, extent, shape, filternorm, filterrad, imlim, resample, url, **kwargs)
   5669                               resample=resample, **kwargs)
   5670 
-> 5671         im.set_data(X)
   5672         im.set_alpha(alpha)
   5673         if im.get_clip_path() is None:

~/anaconda3/lib/python3.7/site-packages/matplotlib/image.py in set_data(self, A)
    688                 or self._A.ndim == 3 and self._A.shape[-1] in [3, 4]):
    689             raise TypeError("Invalid shape {} for image data"
--> 690                             .format(self._A.shape))
    691 
    692         if self._A.ndim == 3:

TypeError: Invalid shape (302, 302, 0) for image data

**Comment by vincent1bt** [comment]
Yes, matplotlib only plots an image if it has a shape of (width, height, channels) where channels must be 3, or in case of a grayscale image its shape must be (width, height). 

The **resample** function should return an image of size (width, height, 1) and we remove the last axis with **np.squeeze** so we can plot it. Try to print the shape of the image returned by the **resample** function and plot it before the **np.squeeze** function to check if that's the problem. 

Maybe the image you are using has a shape of (width, height, 1) or (width, height, 3), if that is the case, then remove the last axis before using it in the **resample** function.
